File: client/src/types/JsonModels/AdminSubscriptions/SubscriptionQueryRequest.ts
import type { JsonModels } from "../../backend";
import type { SubscriptionStatus } from "../Enums/SubscriptionStatus";

export interface SubscriptionQueryRequest extends JsonModels.Common.PagedRequest
{
	search?: string;
	planCode?: string;
	status?: SubscriptionStatus;
	overriddenOnly: boolean;
}

File: server/FitMate.Core/JsonModels/AdminSubscriptions/SubscriptionQueryRequest.cs
using System.ComponentModel.DataAnnotations;
using FitMate.Core.JsonModels.Common;
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.AdminSubscriptions;

public class SubscriptionQueryRequest : PagedRequest
{
    [StringLength(200)]
    public string? Search { get; set; }

    [StringLength(50)]
    public string? PlanCode { get; set; }

    public SubscriptionStatus? Status { get; set; }

    /// <summary>Only users whose plan comes from an administrator override.</summary>
    public bool OverriddenOnly { get; set; }
}
